lib/improver/tests/test_set_up_test_cubes.py

```diff
@@ -0,0 +1,337 @@
+                                           datetime(2017, 12, 1, 14, 0)),
+            datetime(2017, 12, 1, 9, 0))
+        time_coord = coord_dims[0][0]
+        self.assertEqual(iris_time_to_datetime(time_coord)[0],
+                         datetime(2017, 12, 1, 14, 0))
+        self.assertEqual(time_coord.bounds[0][0], time_coord.points[0] - 3600)
+        self.assertEqual(time_coord.bounds[0][1], time_coord.points[0])
+
+    def test_time_bounds_wrong_order(self):
+        """Test time bounds are correctly applied even if supplied in the wrong
+        order"""
+        coord_dims = construct_scalar_time_coords(
+            datetime(2017, 12, 1, 14, 0), (datetime(2017, 12, 1, 14, 0),
+                                           datetime(2017, 12, 1, 13, 0)),
+            datetime(2017, 12, 1, 9, 0))
+        time_coord = coord_dims[0][0]
+        self.assertEqual(iris_time_to_datetime(time_coord)[0],
+                         datetime(2017, 12, 1, 14, 0))
+        self.assertEqual(time_coord.bounds[0][0], time_coord.points[0] - 3600)
+        self.assertEqual(time_coord.bounds[0][1], time_coord.points[0])
+
+    def test_error_invalid_time_bounds(self):
+        """Test an error is raised if the time point is not between the
+        specified bounds"""
+        msg = 'not within bounds'
+        with self.assertRaisesRegex(ValueError, msg):
+            _ = construct_scalar_time_coords(
+                None, (datetime(2017, 12, 1, 13, 0),
+                       datetime(2017, 12, 1, 14, 0)), None)
+
+
+class test_set_up_variable_cube(IrisTest):
+    """Test the set_up_variable_cube base function"""
+
+    def setUp(self):
+        """Set up simple temperature data array"""
+        self.data = (
+            np.linspace(275.0, 284.0, 12).reshape(3, 4).astype(np.float32))
+        self.data_3d = np.array([self.data, self.data, self.data])
+
+    def test_defaults(self):
+        """Test default arguments produce cube with expected dimensions
+        and metadata"""
+        result = set_up_variable_cube(self.data)
+
+        # check type, data and attributes
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.name(), 'air_temperature')
+        self.assertEqual(result.units, 'K')
+        self.assertArrayAlmostEqual(result.data, self.data)
+        self.assertEqual(result.attributes, {})
+
+        # check dimension coordinates
+        self.assertEqual(result.coord_dims("latitude"), (0,))
+        self.assertEqual(result.coord_dims("longitude"), (1,))
+
+        # check scalar time coordinates
+        for time_coord in ["time", "forecast_reference_time",
+                           "forecast_period"]:
+            self.assertEqual(result.coord(time_coord).dtype, np.int64)
+
+        expected_time = datetime(2017, 11, 10, 4, 0)
+        time_point = iris_time_to_datetime(result.coord("time"))[0]
+        self.assertEqual(time_point, expected_time)
+
```

Comment:
I may have missed it, but I don't think this test currently shows that the result for time is a scalar coordinate rather than a single entry dimension coordinate. We could include:
```suggestion
        self.assertFalse(result.coords('time', dim_coords=True))
```

Or alternatively we could simply test that all the dimension coordinates are the ones we expect.